src/components/register/Register/index.jsx

```diff
@@ -0,0 +1,230 @@
+  });
+  const [errors, setErrors] = useState({});
+  const [isTermsOpen, setIsTermsOpen] = useState(false);
+  const handleChange = (key, value) => {
+    const newValues = { ...registerInformation };
+    newValues[key] = value;
+    setRegisterInformation(newValues);
+  };
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (Object.keys(newErrorsObj).length > 0) {
+      setErrors(newErrorsObj);
+      return;
+    }
+    onSubmit(registerInformation);
+  };
+
+  const handleOkTerms = () => {
+    handleChange("isAgreed", true);
+    setIsTermsOpen(false);
+  };
+  const handleCancelTerms = () => {
+    setIsTermsOpen(false);
+  };
+  const termsAndConditionsTranslation = [
+    ...t("register.terms", { returnObjects: true }),
+  ];
+
+  return (
+    <form noValidate onSubmit={handleSubmit}>
+      <Modal
+        title={t("register.termsTitle")}
+        visible={isTermsOpen}
+        onOk={handleOkTerms}
+        onCancel={handleCancelTerms}
+        footer={[
+          <Button key="back" onClick={handleCancelTerms}>
+            {t("register.close")}
+          </Button>,
+          <Button key="submit" type="primary" onClick={handleOkTerms}>
+            {t("register.iAgree")}
+          </Button>,
+        ]}
+      >
+        {t("register.agreeOnTerms", { returnObjects: true })}
+      </Modal>
+      <div className="loginContainer">
+        <div className="loginContainer__loginTitle">{t("register.create")}</div>
+        {error && (
+          <Alert
+            style={{ marginBottom: 10 }}
+            type="error"
+            showIcon
+            message={error}
+          />
+        )}
+        <div className="loginContainer__loginDialog">
+          <div className="loginContainer__loginDialog__inputLabel">
+            {t("register.name")}
+          </div>
+          <div className="loginContainer__loginDialog__input">
+            <Input
+              className={
+                errors.fullName &&
+                "loginContainer__loginDialog__input__hasError"
+              }
+              value={registerInformation.fullName}
+              onChange={(e) => {
+                handleChange("fullName", e.target.value);
+              }}
+            />
+            {errors.fullName && (
+              <div className="loginContainer__loginDialog__errorContainer">
+                {errors.fullName}
+              </div>
+            )}
+          </div>
+          <div className="loginContainer__loginDialog__inputLabel">
+            {t("register.enterAge")}
+          </div>
+          <div className="loginContainer__loginDialog__input">
+            <Input
+              className={
+                errors.age && "loginContainer__loginDialog__input__hasError"
+              }
+              type="number"
+              value={registerInformation.age}
+              onChange={(e) => {
+                handleChange("age", e.target.value);
+              }}
+            />
+            {errors.age && (
```

Comment:
fix this in every PR , i cmomented on one of them about common code

Reply:
I will